src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -91,22 +94,33 @@ public void onPost(String triggeredByUser) {
      * Called when a POST is made.
      */
     public void onPost(final GitHubTriggerEvent event) {
+        if (null == job) {
+            return; // nothing to do
+        }
+
+        Job<?, ?> currentJob = notNull(job, "Job can't be null");
+
         final String pushBy = event.getTriggeredByUser();
         DescriptorImpl d = getDescriptor();
         d.checkThreadPoolSizeAndUpdateIfNecessary();
         d.queue.execute(new Runnable() {
             private boolean runPolling() {
                 try {
-                    StreamTaskListener listener = new StreamTaskListener(getLogFile());
+                    StreamTaskListener listener = new StreamTaskListener(getLogFileForJob(currentJob));
 
                     try {
                         PrintStream logger = listener.getLogger();
+
                         long start = System.currentTimeMillis();
                         logger.println("Started on " + DateFormat.getDateTimeInstance().format(new Date()));
                         if (event.getOrigin() != null) {
                             logger.format("Started by event from %s on %tc%n", event.getOrigin(), event.getTimestamp());
                         }
-                        boolean result = SCMTriggerItems.asSCMTriggerItem(job).poll(listener).hasChanges();
+                        SCMTriggerItem item = SCMTriggerItems.asSCMTriggerItem(currentJob);
+                        if (null == item) {
```

Comment:
As we are updating to newer java, please update null checks to human readable `Objects.isNull/nonNull()` 
(only in modified places)

Reply:
Updated!  I'm just waiting for the checks to make sure nothing changed with the spotbugs

Reply:
I think it can't understand checks, so ignore it.